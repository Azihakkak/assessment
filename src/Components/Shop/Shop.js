import React, { Component } from "react";

import CartView from "../CartView/CartView";

const items = [
  { name: "Shower gel", price: 49.99 },
  { name: "Deodorant", price: 99.99, id: 2 },
];

export default class Shop extends Component {
  constructor() {
    super();
    this.state = {
      selectedItem: "Shower gel",
      itemQty: 1,
      cart: {},
    };
  }

  handleChange = (event) => {
    const { value, name } = event.target;
    this.setState({
      [name]: value,
    });
  };

  handleClick = () => {
    this.setState({
      itemQty: this.state.itemQty + 1,
    });
  };

  handleSubmit = (event) => {
    event.preventDefault();
    const selectedItem = this.state.selectedItem;
    const qty = this.state.itemQty;
    let cart = this.state.cart;

    const newQuantity = cart[selectedItem]
      ? +cart[selectedItem].quantity + +qty
      : qty;
    const item = items.find((item) => item.name === selectedItem);

    cart[selectedItem] = {
      quantity: newQuantity,
      price: item.price,
    };

    this.setState({ ...this.state, cart });
  };

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>
          <label>
            Select your item
            <select
              value={this.state.selectedItem}
              onChange={this.handleChange}
              name="selectedItem"
            >
              <option value="Shower gel">Shower gel</option>
              <option value="Deodorant">Deodorant</option>
            </select>
          </label>
          <div>
            <input
              name="itemQty"
              type="number"
              value={this.state.itemQty}
              onChange={this.handleChange}
            />
          </div>
          <input type="submit" value="Add to cart" />
        </form>
        <CartView cart={this.state.cart} />
        {/* cart.item name cart.quantity cart.qty * price  */}
      </div>
    );
  }
}

// user adds 5 shower gels to the cart
// shopping cart should show 5 shower gels with unit price
// total price should be 249.95
