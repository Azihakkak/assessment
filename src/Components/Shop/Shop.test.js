import React, { Component } from "react";
import Shop from "./Shop";
import renderer from "react-test-renderer";

describe("Shop component", () => {
  test("should render shop component", () => {
    const component = renderer.create(<Shop />);

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("should add shopping gel to the empty cart", () => {
    const component = renderer.create(<Shop />);

    // component.findByName("shopping items");
  });
});
