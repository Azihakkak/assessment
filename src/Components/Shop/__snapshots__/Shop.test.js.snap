// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component should render shop component 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Select your item
      <select
        name="selectedItem"
        onChange={[Function]}
        value="Shower gel"
      >
        <option
          value="Shower gel"
        >
          Shower gel
        </option>
        <option
          value="Deodorant"
        >
          Deodorant
        </option>
      </select>
    </label>
    <div>
      <input
        name="itemQty"
        onChange={[Function]}
        type="number"
        value={1}
      />
    </div>
    <input
      type="submit"
      value="Add to cart"
    />
  </form>
</div>
`;
